### PR TITLE
update library for image component type

### PR DESCRIPTION
**Pull request by yangcao77**
### What does this PR do?:

This PR update the library for the new image component type added for outer loop support. No changes needed for the new group kind `deploy`
Also updates the unit test for it's particular helper function to test image component type. 
The changes need to be made in functional tests in api and library can by tracked via https://github.com/devfile/api/issues/627

### Which issue(s) this PR fixes:

https://github.com/devfile/api/issues/626

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [x] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: pkg/devfile/parser/data/v2/common/component_helper.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ func GetComponentType(component v1.Component) (v1.ComponentType, error) {
 		return v1.KubernetesComponentType, nil
 	case component.Openshift != nil:
 		return v1.OpenshiftComponentType, nil
+	case component.Image != nil:
+		return v1.ImageComponentType, nil
 	case component.Custom != nil:
 		return v1.CustomComponentType, nil
```
Comment (kim-tsao) [inline pkg/devfile/parser/data/v2/common/component_helper.go:36]: Do we need to add an IsImage() check for testing?

File: pkg/devfile/parser/data/v2/common/component_helper_test.go [hunk 2]
```diff
@@ -150,6 +150,16 @@ func TestGetComponentType(t *testing.T) {
...
+		{
+			name: "Image component",
+			component: v1.Component{
+				Name: "name",
+				ComponentUnion: v1.ComponentUnion{
+					Image: &v1.ImageComponent{},
+				},
+			},
+			componentType: v1.ImageComponentType,
+		},
```
Comment (kim-tsao) [inline pkg/devfile/parser/data/v2/common/component_helper_test.go:162]: Similar to the comment above, do we need to add  a TestIsImage() test for this new component?

File: pkg/devfile/parser/data/v2/common/component_helper.go (hunk 1 above)
Comment (yangcao77) [reply pkg/devfile/parser/data/v2/common/component_helper.go:36]: If you check the file, we do not have this helper function for all component types, so I did not add `IsImage()`, and we can always add more helper functions if there is a request from consumers. 
I'm fine with adding `isImage` if you think it should be added

Comment (kim-tsao) [reply pkg/devfile/parser/data/v2/common/component_helper.go:36]: Yes, we are missing that check for other components as well and I wasn't sure if it was an oversight or not.  I'm ok with not adding it if there is no immediate value in testing it.

**Review by kim-tsao — APPROVED** [review]

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**